GP11 : MAC functions fix/panic reason

cedric-chaumont-st-dev opened the pull request:
Signed-off-by: Cedric Chaumont <cedric.chaumont@st.com>

cedric-chaumont-st-dev commented:
Updated.

pascal-brand-st-dev commented on lib/libutee/tee_api_operations.c:
```diff
@@ -860,75 +860,141 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle op,
+	case TEE_ALG_DES3_CBC_MAC_NOPAD:
+	case TEE_ALG_DES3_CBC_MAC_PKCS5:
+		if (IV == NULL && IVLen != 0)
+			TEE_Panic(0);
```
I would not panic as IV and IVLen are never used in the current algorithms

cedric-chaumont-st-dev replied:
Current algo listed use IV except CMAC ( I will remove it). Do you agree ?

pascal-brand-st-dev commented on lib/libutee/tee_api_operations.c:
```diff
@@ -860,75 +860,141 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle op,
+	case TEE_ALG_AES_CMAC:
+	case TEE_ALG_DES_CBC_MAC_PKCS5:
+	case TEE_ALG_DES3_CBC_MAC_NOPAD:
+	case TEE_ALG_DES3_CBC_MAC_PKCS5:
```
Missing HMAC algo.

cedric-chaumont-st-dev replied:
HAMC/CMAC algos do not use IV. Do you agree ?

pascal-brand-st-dev commented on lib/libutee/tee_api_operations.c:
```diff
@@ -860,75 +860,141 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle op,
+		break;
+
+	default:
+		iv = NULL;
```
I would panic as we should never reach default.

cedric-chaumont-st-dev replied:
Even if currently not implemented, algos that could use IV should have coherent parameters. Do you agree ?

cedric-chaumont-st-dev commented:
OK I will remove the check. IV=0 with CBC-MAC and keep the note.

cedric-chaumont-st-dev commented:
Updated. Rebased.

pascal-brand-st-dev commented on lib/libutee/tee_api_operations.c:
```diff
@@ -860,75 +860,121 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle op,
 	if (operation->info.operationClass != TEE_OPERATION_MAC)
 		TEE_Panic(0);
-	res = utee_hash_init(operation->state, IV, IVLen);
+	/* Note : iv and ivlen are never used in current implementation */
```
Can you add:

    This is why coherent values of IV and IVLen are not checked

pascal-brand-st-dev commented:
Apart from a minor comment:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

cedric-chaumont-st-dev commented:
Rebased.

cedric-chaumont-st-dev commented:
Rebased.Squashed. Tested (STM, ARM Juno).
28701 subtests of which 0 failed
687 test cases of which 0 failed
0 test case was skipped
TEE test application done!
